File: src/main/java/tfg/proyecto/TFG/modelo/TarjetaBancaria.java
package tfg.proyecto.TFG.modelo;

import java.math.BigDecimal;
import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
public class TarjetaBancaria {
	
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombreTitular;
    @Column(length = 16)
    private String nTarjeta;
    private LocalDate fechaCaducidad;
    @Column(length = 4)
    private String cvv;  
    private BigDecimal saldo;

}

File: appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/model/TarjetaBancaria.kt
package com.example.appmovilshowpass.model

import java.math.BigDecimal
import java.time.LocalDate

data class TarjetaBancaria(
    val id: Long = 0,
    val nombreTitular: String = "",
    val fechaCaducidad: LocalDate,
    val cvv: String = "",
    val saldo: BigDecimal,
    val ntarjeta: String = "",

    )